cpp/include/cudf/detail/scatter.cuh

```diff
@@ -299,11 +420,16 @@ std::unique_ptr<table> scatter(
                                                                  mr);
                  });
 
-  auto gather_map = scatter_to_gather(
-    updated_scatter_map_begin, updated_scatter_map_end, target.num_rows(), stream);
-
-  gather_bitmask(source, gather_map.begin(), result, gather_bitmask_op::PASSTHROUGH, stream, mr);
-
+  // We still need to call `gather_bitmask` even when the source columns are not nullable,
+  // as if the target has null_mask, that null_mask needs to be updated after scattering
+  auto const nullable =
+    std::any_of(source.begin(), source.end(), [](auto const& col) { return col.nullable(); }) or
+    std::any_of(target.begin(), target.end(), [](auto const& col) { return col.nullable(); });
```

Comment:
```suggestion
    std::any_of(source.cbegin(), source.cend(), [](auto const& col) { return col.nullable(); }) or
    std::any_of(target.cbegin(), target.cend(), [](auto const& col) { return col.nullable(); });
```

Reply:
Unfortunately `source` and `target` are `table_view`, which doesn't have const iterators.